File: app/src/main/java/net/bloople/manga/MatchWidthTransformation.java
package net.bloople.manga;

import android.graphics.Bitmap;

import androidx.annotation.NonNull;

import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool;
import com.bumptech.glide.load.resource.bitmap.BitmapTransformation;
import com.bumptech.glide.load.resource.bitmap.TransformationUtils;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;

public class MatchWidthTransformation extends BitmapTransformation {
    private static final String ID = "net.bloople.manga.MatchWidthTransformation";
    private static final byte[] ID_BYTES = ID.getBytes(StandardCharsets.UTF_8);

    @Override
    public Bitmap transform(@NonNull BitmapPool pool, Bitmap toTransform, int outWidth, int outHeight) {
        int targetWidth = outWidth;
        int targetHeight = toTransform.getHeight();

        return TransformationUtils.fitCenter(pool, toTransform, targetWidth, targetHeight);
    }

    @Override
    public boolean equals(Object o) {
        return o instanceof MatchWidthTransformation;
    }

    @Override
    public int hashCode() {
        return ID.hashCode();
    }

    @Override
    public void updateDiskCacheKey(MessageDigest messageDigest) {
        messageDigest.update(ID_BYTES);
    }
}

File: app/src/main/java/net/bloople/manga/MatchWidthTransformation.kt
package net.bloople.manga

import com.bumptech.glide.load.resource.bitmap.BitmapTransformation
import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool
import android.graphics.Bitmap
import com.bumptech.glide.load.resource.bitmap.TransformationUtils
import java.nio.charset.StandardCharsets
import java.security.MessageDigest

class MatchWidthTransformation : BitmapTransformation() {
    public override fun transform(pool: BitmapPool, toTransform: Bitmap, outWidth: Int, outHeight: Int): Bitmap {
        val targetHeight = toTransform.height
        return TransformationUtils.fitCenter(pool, toTransform, outWidth, targetHeight)
    }

    override fun equals(other: Any?): Boolean {
        return other is MatchWidthTransformation
    }

    override fun hashCode(): Int {
        return ID.hashCode()
    }

    override fun updateDiskCacheKey(messageDigest: MessageDigest) {
        messageDigest.update(ID_BYTES)
    }

    companion object {
        private const val ID = "net.bloople.manga.MatchWidthTransformation"
        private val ID_BYTES = ID.toByteArray(StandardCharsets.UTF_8)
    }
}